### ISSUE TITLE
`derivative` should lower order of the taylor series

### ISSUE THREAD

**Issue by Keno** [JuliaDiff/TaylorSeries.jl#268]
```
julia> Taylor1(Float64, 5)
 1.0 t + 𝒪(t⁶)

julia> derivative(ans)
 1.0 + 𝒪(t⁶)
```

I'd have expected `𝒪(t⁵) terms for the derivative. As is, this implies that the t⁵ term is zero in the derivative, which is not the case.

**Comment by dpsanders** [comment]
I agree that this seems incorrect.

**Comment by lbenet** [comment]
Thanks for reporting this, @Keno. 

I am actually addressing this, together with other cases (c.f. #230), but haven't had the time to complete it.

**Comment by lbenet** [comment]
Addressed in #248

**Comment by lbenet** [comment]
Closed by #248
